rebase: update kubernetes deps to 1.27.1

riya-singhal31 opened the pull request:
As the k8s 1.27 is released, updating the do dependencies to 1.27.

riya-singhal31 committed:
rebase: update kubernetes deps to 1.27

Signed-off-by: riya-singhal31 <rsinghal@redhat.com>

Madhu-1 commented:
@riya-singhal31 please rebase the PR

riya-singhal31 commented:
> @riya-singhal31 please rebase the PR

Rebased, thanks!

Madhu-1 commented on go.mod:
```diff
@@ -35,21 +35,24 @@ require (
 	//
-	k8s.io/api v0.26.3
-	k8s.io/apimachinery v0.26.3
+	k8s.io/api v0.27.0
```
we have kubernetes 1.27.1 release , instead of updating to older release lets update to new release.

riya-singhal31 replied:
Yes, sure.
Updated to 1.27.1

Madhu-1 commented:
https://github.com/ceph/ceph-csi/blob/f0e6ee047ee388a7299d70dfbb5784546540725f/go.mod#L183-L186 time to remove this as we are moving to 1.27?

riya-singhal31 commented:
> https://github.com/ceph/ceph-csi/blob/f0e6ee047ee388a7299d70dfbb5784546540725f/go.mod#L183-L186
> 
> time to remove this as we are moving to 1.27?

Removed the other two, leaving github.com/portworx/sched-ops => github.com/portworx/sched-ops v0.20.4-openstorage-rc3 
as this is still required.

Madhu-1 reviewed (changes requested):
Please fix the CI failures

Madhu-1 commented:
@riya-singhal31 please rebase the PR

riya-singhal31 commented:
> @riya-singhal31 please rebase the PR

Rebased, thanks!

riya-singhal31 commented:
It seems like a bug in operator-sdk client.go https://github.com/operator-framework/operator-sdk/issues/6396

riya-singhal31 committed:
cleanup: ErrWaitTimeout is deprecated in k8s 1.27

replaced ErrWaitTimeout with Interrupted

Signed-off-by: riya-singhal31 <rsinghal@redhat.com>

nixpanic committed:
ci: new Ceph packages require /etc/selinux/config

It seems `ceph-selinux` now needs `/etc/selinux/config`, otherwise
installing will fail.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
rebase: update controller-runtime to current `main`

Kubernetes 1.27 is not compatible with any released controller-runtime
version. A commit that addresses the issue is only available in the
`main` branch of the project for now, so rebasing to that version.

See-also: kubernetes-sigs/controller-runtime#2223
Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
cleanup: remove unused import "errors"

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
e2e: pass context when creating a new Namespace

Signed-off-by: Niels de Vos <ndevos@ibm.com>

riya-singhal31 commented:
Thanks for the commits @nixpanic. I'll take a look and fix the rest of CIs.

Madhu-1 commented:
@riya-singhal31 can you please refresh the PR?

riya-singhal31 commented:
> @riya-singhal31 can you please refresh the PR?

Thanks, rebased.
Fixing the linters.
